src/test/java/net/snowflake/ingest/streaming/internal/DataValidationUtilTest.java

```diff
@@ -277,74 +264,51 @@ public void testValidateAndPareTimestampTz() {
-    for (Object input : goodIntegersValue10) {
-      Assert.assertEquals(
-          new BigInteger("10"), DataValidationUtil.validateAndParseBigInteger(input));
-    }
-    Assert.assertEquals(
-        new BigInteger("-1000"), DataValidationUtil.validateAndParseBigInteger("-1e3"));
-
-    Assert.assertEquals(
-        BigInteger.valueOf(10).pow(37),
-        DataValidationUtil.validateAndParseBigInteger(BigInteger.valueOf(10).pow(37)));
-    Assert.assertEquals(
-        BigInteger.valueOf(-1).multiply(BigInteger.valueOf(10).pow(37)),
-        DataValidationUtil.validateAndParseBigInteger(
-            BigInteger.valueOf(-1).multiply(BigInteger.valueOf(10).pow(37))));
-
-    // Expect errors
-    // Too big
-    expectError(
-        ErrorCode.INVALID_ROW,
-        DataValidationUtil::validateAndParseBigInteger,
-        BigInteger.valueOf(10).pow(38));
-    // Too small
-    expectError(
-        ErrorCode.INVALID_ROW,
-        DataValidationUtil::validateAndParseBigInteger,
-        BigInteger.valueOf(-1).multiply(BigInteger.valueOf(10).pow(38)));
-    // Decimal
-    expectError(ErrorCode.INVALID_ROW, DataValidationUtil::validateAndParseBigInteger, 1.1D);
-    expectError(ErrorCode.INVALID_ROW, DataValidationUtil::validateAndParseBigInteger, 1.1F);
-    expectError(ErrorCode.INVALID_ROW, DataValidationUtil::validateAndParseBigInteger, "1.1");
-  }
-
   @Test
   public void testValidateAndParseBigDecimal() {
-    Assert.assertEquals(new BigDecimal("1"), DataValidationUtil.validateAndParseBigDecimal("1"));
-    Assert.assertEquals(
-        new BigDecimal("1000").toBigInteger(),
-        DataValidationUtil.validateAndParseBigDecimal("1e3").toBigInteger());
-    Assert.assertEquals(
-        new BigDecimal("-1000").toBigInteger(),
-        DataValidationUtil.validateAndParseBigDecimal("-1e3").toBigInteger());
-    Assert.assertEquals(
-        new BigDecimal("1").toBigInteger(),
-        DataValidationUtil.validateAndParseBigDecimal("1e0").toBigInteger());
-    Assert.assertEquals(
-        new BigDecimal("-1").toBigInteger(),
-        DataValidationUtil.validateAndParseBigDecimal("-1e0").toBigInteger());
-    Assert.assertEquals(
-        new BigDecimal("123").toBigInteger(),
-        DataValidationUtil.validateAndParseBigDecimal("1.23e2").toBigInteger());
-    Assert.assertEquals(new BigDecimal("1"), DataValidationUtil.validateAndParseBigDecimal(1));
-    Assert.assertEquals(new BigDecimal("1.0"), DataValidationUtil.validateAndParseBigDecimal(1D));
-    Assert.assertEquals(new BigDecimal("1"), DataValidationUtil.validateAndParseBigDecimal(1L));
-    Assert.assertEquals(new BigDecimal("1.0"), DataValidationUtil.validateAndParseBigDecimal(1F));
-    Assert.assertEquals(
-        BigDecimal.valueOf(10).pow(37),
-        DataValidationUtil.validateAndParseBigDecimal(BigDecimal.valueOf(10).pow(37)));
-    Assert.assertEquals(
+    assertEquals(new BigDecimal("1"), validateAndParseBigDecimal("1"));
+    assertEquals(
```

Comment:
does it make sense to declare a list of cases for tests with many repetitions?
Like:
```
cases = {
  {"1000", "1e3"},
  ...
}

test() {
  for case in cases {
    input = case.1;
    expected = case.2;
    assertEquals(new BigDecimal(input).toBigInteger(), validateAndParseBigDecimal(expected).toBigInteger());
  }
}
```

Reply:
I agree, it makes sense. Let's discuss it on our 1-1 and I can change it later for all test cases.